dev-java/gson: add 2.10.1

vaukai opened the pull request:
Signed-off-by: Volkmar W. Pogatzki <gentoo@pogatzki.net>

fordfrog commented:
looking at the previous ebuild, you can use `jdk-17` in build deps and `jre-8` in runtime deps if the package compiles without issues when tests are not triggered during the build.

vaukai commented:
> looking at the previous ebuild, you can use `jdk-17` in build deps and `jre-8` in runtime deps if the package compiles without issues when tests are not triggered during the build.

There is only one testclass which wants `source 17`. I removed it and added a comment in the `ebuild`.
